Dockerfile

```diff
@@ -0,0 +1,33 @@
+FROM osrf/ros:humble-desktop-full
+
+# Install dependencies (taken from devcontainer dockerfile)
+RUN DEBIAN_FRONTEND=noninteractive apt-get update -q && \
+    apt-get update -q && \
+    apt-get install -yq --no-install-recommends \
+    curl \
+    wget \
+    python3-pip \
+    python-is-python3 \
+    python3-argcomplete \
+    python3-colcon-common-extensions \
+    python3-colcon-mixin \
+    python3-rosdep \
+    libpython3-dev \
+    python3-vcstool \
+    ros-humble-tl-expected && \
+    rm -rf /var/lib/apt/lists/*
+
+# Create ROS workspace
+WORKDIR /ros_ws/src
+
+# Clone driver code
+RUN git clone https://github.com/bdaiinstitute/spot_ros2.git .
+RUN git submodule update --init --recursive
+
+# Run install script
+RUN /ros_ws/src/install_spot_ros2.sh
+
+# Build packages with Colcon
+WORKDIR /ros_ws/
+RUN . /opt/ros/humble/setup.sh && \
+    colcon build --symlink-install --packages-ignore proto2ros proto2ros_tests
```

Comment:
@ryan-roche consider installing to a known location and ensuring the binary overlay gets sourced by updating the image entrypoint. I still think a `--merge-install` is the most appropriate.

Reply:
Also, how would you support a user that needs to temporarily patch `spot_ros2`?

Reply:
I'm not sure how we would be able to support that workflow in a docker container like this. Wouldn't a `--merge-install` only make that more difficult?

We could run a `--merge-install` to /opt/ros/noetic so that the driver is included in the underlay regardless of where the user puts their project inside the container, but that would bring up even more issues should they need to patch the driver themselves.

I was operating on the assumption that the user would just work in the already-created `ros_ws` directory and simply rebuild `spot_ros2` if they had to patch it. Precompiling the package was more for making the image a somewhat turn-key deployment

Reply:
Circling back. Perhaps document `/ros_ws` as the well-known workspace inside the container, so that users can bind mount `spot_ros2` or their own repositories while developing. In that case, a `--merge-install` will spare you the broken links if some `spot_ros2` sources go away in that process.